### PR TITLE
extended pppos_client (IDFGH-8392)

### PR DESCRIPTION
**Pull request by diplfranzhoepfinger**
extended pppos_client

### CODE REVIEW

**Review by franz-ms-muc — APPROVED** [review]

**Comment by diplfranzhoepfinger** [comment]
must do same as in #120

**Review by franz-ms-muc — APPROVED** [review]

File: components/esp_modem/examples/pppos_client/main/pppos_client_main.c [hunk 1]
```diff
@@ -159,6 +175,16 @@ void app_main(void)
...
+    if(dte_config.uart_config.flow_control == ESP_MODEM_FLOW_CONTROL_HW)
+    {
+        esp_err_t err = esp_modem_set_flow_control(dce, 2, 2);  //2/2 means HW Flow Control.
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Failed to set the set_flow_control mode");
+            return;
+        }
+        ESP_LOGI(TAG, "set_flow_control OK");
+    } else {
+    }
```
Comment (david-cermak) [inline components/esp_modem/examples/pppos_client/main/pppos_client_main.c:187]:
```suggestion
```

File: components/esp_modem/examples/pppos_client/main/pppos_client_main.c [hunk 2]
```diff
@@ -159,6 +175,16 @@ void app_main(void)
     esp_modem_dce_t *dce = esp_modem_new(&dte_config, &dce_config, esp_netif);
 #endif
     assert(dce);
+    if(dte_config.uart_config.flow_control == ESP_MODEM_FLOW_CONTROL_HW)
+    {
```
Comment (david-cermak) [inline components/esp_modem/examples/pppos_client/main/pppos_client_main.c:179]:
```suggestion
    if(dte_config.uart_config.flow_control == ESP_MODEM_FLOW_CONTROL_HW) {
```

**Review by david-cermak — APPROVED** [review]

**Comment by david-cermak** [comment]
Accepted via https://github.com/espressif/esp-protocols/pull/165 (merged as ebc36a3519ce8633ead394b0f15cb90780d6e397)

**Pull request closed without merging**